src/smt/smt_engine.cpp

```diff
@@ -1007,71 +957,47 @@ theory::TheoryModel* SmtEngine::getAvailableModel(const char* c) const
   return m;
 }
 
-void SmtEngine::notifyPushPre() { processAssertionsInternal(); }
+void SmtEngine::notifyPushPre() { d_smtSolver->processAssertions(*d_asserts); }
+
 void SmtEngine::notifyPushPost()
 {
   TimerStat::CodeTimer pushPopTimer(d_stats->d_pushPopTime);
-  d_propEngine->push();
+  PropEngine* pe = getPropEngine();
+  Assert(pe != nullptr);
+  pe->push();
```

Comment:
This is a pattern I've seen a few times now.
I'd suggest the following instead, as it seems more natural to me:
```suggestion
  Assert(getPropEngine() != nullptr);
  getPropEngine()->push();
```